snuba/query/allocation_policies/concurrent_rate_limit.py

```diff
@@ -80,6 +81,11 @@ def _is_within_rate_limit(
             rate_limit_prefix,
             self.get_config_value("max_query_duration_s"),
         )
+        return rate_limit_stats
+
+    def _is_within_rate_limit(
```

Comment:
I think the easier thing would be to return the rate_limit_stats from the original _is_within_rate_limit_function, that way you only have to change the call site of the other policies, not the actual logic

Reply:
To me, it doesn't seem semantically correct to have `_is_within_rate_limit` return `rate_limit_stats`. As the function name implies, it feels like it should just return `True/False` and the explanation of why it is or isn't within the rate limits. That's why I decided to change split them and add a `_get_rate_limit_stats` function. What do you think?

Reply:
I'm not too attached to my opinion but here's what I'll say:

Before this change, every policy would just have to call `_is_within_rate_limit` now every policy has to call `_get_rate_limit_stats` and `_is_within_rate_limit` 

You can't really call `_is_within_rate_limit` without getting the rate limit information first, these things will never be called separately. I'm thinking of the `rate_limit_stats` as another detailed piece of the explanation as to why the function returned True/False and that's why I consider it the more reasonable thing to surface that data alongside the explanation.

Reply:
> You can't really call _is_within_rate_limit without getting the rate limit information first, these things will never be called separately. I'm thinking of the rate_limit_stats as another detailed piece of the explanation

Thats fair. Updated, thanks volo!